Controllers/SearchController.cs

```diff
@@ -31,9 +40,16 @@ private async Task<IActionResult> SearchAsync(ProviderType providerType, string
             if (string.IsNullOrWhiteSpace(query))
                 return BadRequest("Missing query parameter.");
 
+            if (_cache != null &&
```

Comment:
If it's already in DI then no need to check for null.

Reply:
If you want to alaways inject in the dependencie ok, but see the startup you have a if to inject this service in the DI